portfolio/src/main/java/com/google/sps/servlets/ReadCommentsServlet.java

```diff
@@ -0,0 +1,81 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.sps.commentart.Comment;
+import com.google.sps.commentart.RepliesManager;
+import com.google.sps.commentart.Reply;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Returns a list of comments that are posted on CommentArt. */
+@WebServlet("/read_comments")
+public class ReadCommentsServlet extends HttpServlet {
+  private final Gson gson = new Gson();
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query commentQuery = new Query("CommentWithSenti").addSort("timestamp", SortDirection.ASCENDING);
+    PreparedQuery results = datastore.prepare(commentQuery);
+
+    List<Comment> comments = new ArrayList<>();
+    for (Entity commentEntity : results.asIterable()) {
+      comments.add(toComment(commentEntity));
+    }
+    String json = gson.toJson(comments);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  private Comment toComment(Entity commentEntity) {
```

Comment:
These `to` methods should not be in the scope of ReadCommentsServlet, instead, they should be the knowledge of each data class.

Reply:
Okay. Thanks!